src/main/java/org/candlepin/subscriptions/tally/TallySnapshotController.java

```diff
@@ -119,16 +120,20 @@ public void produceSnapshotsForAccounts(List<String> accounts) {
     @Timed("rhsm-subscriptions.snapshots.single.hourly")
     public void produceHourlySnapshotsForAccount(String accountNumber, OffsetDateTime startDateTime,
         OffsetDateTime endDateTime) {
-
         try {
-            Map<OffsetDateTime, AccountUsageCalculation> accountCalcs = retryTemplate.execute(
+            Map<OffsetDateTime, AccountUsageCalculation>  accountCalcs = retryTemplate.execute(
                 context -> metricUsageCollector.collect(accountNumber, startDateTime, endDateTime));
+
+            var applicableUsageCalculations = accountCalcs.entrySet().stream()
+                .filter(TallySnapshotController::isCombiningRollupStrategySupported)
```

Comment:
I would prefer to see a strongly typed variable here instead, but it's a minor point.

Reply:
The variable is still strongly typed.  `var` in java lets the type be inferred by the compiler, and is just a convenience/readability feature.  It's the equivalent of below, just less noisy and easier to read in my opinion.  Easily one of the main reasons I pushed upgrading to Java 10 so much.

```java
Map<OffsetDateTime, AccountUsageCalculation> applicableUsageCalculations = accountCalcs.entrySet().stream()
    .filter(TallySnapshotController::isCombiningRollupStrategySupported)
    .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
```

@kahowell, thoughts on using `var` or not?

Reply:
I find var more readable here + Intellij shows the type, so I personally prefer `var` also. Let's see what @mstead thinks as well.

Reply:
Though I will say, I wouldn't bat an eyelash at explicit `Map` either.

Reply:
Might be best to leave me out of this one, because I think the whole var thing is foolish for various reasons. :)

That being said, I really don't have a strong opinion one way or the other. I'm Ok with both.

Reply:
Perhaps @awood can weigh in. :)

Reply:
I prefer `var` for throw away stuff that's just going to be used like once or twice in close proximity to the declaration or when the type is going to be totally unambigious (e.g. `var myThing = new Thing()`).  It's also awesome for when you have to use a fully qualified class name if you have imports that clash on name (e.g. DTO classes and DB model classes).

Overall, I favor it when it is used judiciously.  The guidelines [here](https://openjdk.java.net/projects/amber/LVTIstyle.html) are intelligent and I'd be happy to use `var` in accordance with them.